pandas/plotting/_core.py

```diff
@@ -597,103 +397,227 @@ def boxplot_frame_groupby(grouped, subplots=True, column=None, fontsize=None,
+                ('use_index', True), ('title', None), ('grid', None),
+                ('legend', False), ('style', None), ('logx', False),
+                ('logy', False), ('loglog', False), ('xticks', None),
+                ('yticks', None), ('xlim', None), ('ylim', None),
+                ('rot', None), ('fontsize', None), ('colormap', None),
+                ('table', False), ('yerr', None), ('xerr', None),
+                ('label', None), ('secondary_y', False)]
+        elif isinstance(data, ABCDataFrame):
+            arg_def = [
+                ('x', None), ('y', None), ('kind', 'line'), ('ax', None),
+                ('subplots', False), ('sharex', None), ('sharey', False),
+                ('layout', None), ('figsize', None), ('use_index', True),
+                ('title', None), ('grid', None), ('legend', True),
+                ('style', None), ('logx', False), ('logy', False),
+                ('loglog', False), ('xticks', None), ('yticks', None),
+                ('xlim', None), ('ylim', None), ('rot', None),
+                ('fontsize', None), ('colormap', None), ('table', False),
+                ('yerr', None), ('xerr', None), ('secondary_y', False),
+                ('sort_columns', False)]
         else:
-            raise ValueError("plot kind %r can only be used for data frames"
-                             % kind)
-
-    elif kind in _series_kinds:
-        if isinstance(data, ABCDataFrame):
-            if y is None and subplots is False:
-                msg = "{0} requires either y column or 'subplots=True'"
-                raise ValueError(msg.format(kind))
-            elif y is not None:
-                if is_integer(y) and not data.columns.holds_integer():
-                    y = data.columns[y]
-                # converted to series actually. copy to not modify
-                data = data[y].copy()
-                data.index.name = y
-        plot_obj = klass(data, subplots=subplots, ax=ax, kind=kind, **kwds)
-    else:
-        if isinstance(data, ABCDataFrame):
+            return TypeError(('Called plot accessor for type {}, expected '
+                              'Series or DataFrame').format(
+                                  type(data).__name__))
+
+        pos_args = {name: value for value, (name, _) in zip(args, arg_def)}
+        kwargs = dict(arg_def, **pos_args, **kwargs)
+
+        x = kwargs.pop('x', None)
+        y = kwargs.pop('y', None)
+        kind = kwargs.pop('kind', 'line')
+        return x, y, kind, kwargs
+
+    def __call__(self, *args, **kwargs):
+        x, y, kind, kwargs = self._get_call_args(self._parent, args, kwargs)
+
+        kind = self._kind_aliases.get(kind, kind)
+        if kind not in self._all_kinds:
+            raise ValueError('{} is not a valid plot kind'.format(kind))
+
+        plot_backend = _get_plot_backend()
+        data = self._parent.copy()
+
+        if isinstance(data, pandas.core.dtypes.generic.ABCSeries):
+            kwargs['new_plot'] = True
```

Comment:
This sounds a bit strange (for Series I except to get *no* new plot, so False), so should the meaning of True/False be the other way around?

Reply:
yes, you're right, I started to implement for `new_plot`, and changed it later and forgot to rename to `reuse_plot`